sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -125,20 +125,34 @@ package object util extends Logging {
 
   // Replaces attributes, string literals, complex type extractors with their pretty form so that
   // generated column names don't contain back-ticks or double-quotes.
-  def usePrettyExpression(e: Expression): Expression = e transform {
-    case a: Attribute => new PrettyAttribute(a)
-    case Literal(s: UTF8String, StringType) => PrettyAttribute(s.toString, StringType)
-    case Literal(v, t: NumericType) if v != null => PrettyAttribute(v.toString, t)
-    case Literal(null, dataType) => PrettyAttribute("NULL", dataType)
-    case e: GetStructField =>
-      val name = e.name.getOrElse(e.childSchema(e.ordinal).name)
-      PrettyAttribute(usePrettyExpression(e.child).sql + "." + name, e.dataType)
-    case e: GetArrayStructFields =>
-      PrettyAttribute(usePrettyExpression(e.child) + "." + e.field.name, e.dataType)
-    case r: RuntimeReplaceable =>
-      PrettyAttribute(r.mkString(r.exprsReplaced.map(toPrettySQL)), r.dataType)
-    case c: CastBase if !c.getTagValue(Cast.USER_SPECIFIED_CAST).getOrElse(false) =>
-      PrettyAttribute(usePrettyExpression(c.child).sql, c.dataType)
+  def usePrettyExpression(
+      e: Expression, existsNonSQLExpression: Boolean = false): Expression = e match {
+    case e: NonSQLExpression =>
+      e.withNewChildren(e.children.map(child => usePrettyExpression(child, true)))
```

Comment:
Should we handle this in `NonSQLExpression.sql`?

Reply:
Not need. It's only apply in `toPrettySQL`. For other case which is not invoked by `toPrettySQL` (some error msg), we should keep the origin info.